diff -Naur a/pgraph/src/Pgrapher.cxx b/pgraph/src/Pgrapher.cxx
--- a/pgraph/src/Pgrapher.cxx	2018-05-15 13:30:32.000000000 -0500
+++ b/pgraph/src/Pgrapher.cxx	2018-05-15 13:35:17.598055917 -0500
@@ -28,7 +28,7 @@
     std::string node = jone["node"].asString();
 
     // We should NOT be the one creating this component.
-    auto nptr = Factory::find_maybe_tn<INode>(node);
+    auto nptr = WireCell::Factory::find_maybe_tn<INode>(node);
     if (!nptr) {
         std::cerr << "Pgrapher: failed to get node " << node << "\n";
         THROW(ValueError() << errmsg{"failed to get node"});
diff -Naur a/util/test/Utils.h b/util/test/Utils.h
--- a/util/test/Utils.h	2018-05-15 13:30:36.000000000 -0500
+++ b/util/test/Utils.h	2018-05-15 13:35:58.402728219 -0500
@@ -1,5 +1,5 @@
 #ifndef WIRECELLTEST_UTILS
-#define WIRECELLTEST_Utils
+#define WIRECELLTEST_UTILS
 
 #include "TStyle.h"
 #include "TColor.h"
