diff -Naur wire-cell-build/util/test/test_median.cxx wire-cell-build/util/test/test_median.cxx
--- wire-cell-build/util/test/test_median.cxx	2017-04-05 11:12:11.000000000 -0500
+++ wire-cell-build/util/test/test_median.cxx	2017-04-05 11:56:38.544661591 -0500
@@ -2,6 +2,7 @@
 #include "WireCellUtil/Waveform.h"
 
 #include <iostream>
+#include <random>
 using namespace std;
 using namespace WireCell::Waveform;
 
